packages/app-config-writer/src/variants-config/utils.ts

```diff
@@ -0,0 +1,145 @@
+import { FileName, readUi5Yaml } from '@sap-ux/project-access';
+import { MiddlewareConfigs } from '../types';
+import { stringify } from 'querystring';
+import type { Package } from '@sap-ux/project-access';
+import type { CustomMiddleware, UI5Config } from '@sap-ux/ui5-config';
+import type { PreviewConfigOptions, FioriToolsDeprecatedPreviewConfig } from '../types';
+import { basename } from 'path';
+
+/**
+ * Gets the preview middleware form the yamlConfig or provided path.
+ * The middleware can either be named 'fiori-tools-preview' or 'preview-middleware'.
+ *
+ * @param yamlConfig - the yaml configuration to use; if not provided, the file will be read with the provided basePath and filename
+ * @param basePath - path to project root, where ui5.yaml is located
+ * @param filename - name of the ui5 yaml file to read from basePath; default is 'ui5.yaml'
+ * @returns preview middleware configuration if found<br>
+ * Rejects if neither yamlConfig nor basePath is provided or if the file can't be read
+ */
+export async function getPreviewMiddleware(
```

Comment:
If a function is marked as `async` then you don't need to wrap errors into a `Promise.reject`, you can just throw the error.
With that in mind, I think this function can be simplified.

Reply:
fixed in bff79e66c647e99f33bfef00a58bc2d82eba2999